pkg/newmapper/cronjob.go

```diff
@@ -0,0 +1,97 @@
+package newmapper
+
+import (
+	"strings"
+	"time"
+
+	"github.com/leanix/leanix-k8s-connector/pkg/kubernetes"
+	"github.com/leanix/leanix-k8s-connector/pkg/mapper"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+func GetCronJobs(clusterName string, namespaces *corev1.NamespaceList, kubernetesAPI *kubernetes.API) ([]mapper.KubernetesObject, error) {
+	var allCronJobs []mapper.KubernetesObject
+	for _, namespace := range namespaces.Items {
+		cronJobs, err := kubernetesAPI.CronJobs(namespace.Name)
+		if err != nil {
+			return nil, err
+		}
+		mappedCronJobs, err := MapCronJobs(clusterName, cronJobs)
+		if err != nil {
+			return nil, err
+		}
+		allCronJobs = append(allCronJobs, mappedCronJobs...)
+
+	}
+	return allCronJobs, nil
+}
+
+func MapCronJobs(clusterName string, cronJobs *batchv1.CronJobList) ([]mapper.KubernetesObject, error) {
+	var groupedCronJobs []mapper.KubernetesObject
+	for _, cronJob := range cronJobs.Items {
+		cronJob.ClusterName = clusterName
+		cronJobArtifact, err := CronJobSoftwareArtifact(cronJob)
+		mappedCronJob, err := CronJobDataMapping(cronJob)
+		if err != nil {
+			return nil, err
+		}
+		groupedCronJobs = append(groupedCronJobs, *mappedCronJob, *cronJobArtifact)
+	}
+	return groupedCronJobs, nil
+}
+
+func CronJobSoftwareArtifact(CronJob batchv1.CronJob) (*mapper.KubernetesObject, error) {
+	var CronJobData map[string]interface{}
+	CronJobData = make(map[string]interface{})
+	CronJobId := CronJob.Namespace + "_" + CronJob.Name
+	CronJobData["clusterName"] = CronJob.ClusterName
+	CronJobData["name"] = CronJob.Namespace + ":" + CronJob.Name
+	CronJobData["category"] = "Microservice"
+
+	return &mapper.KubernetesObject{
+		ID:   CronJobId,
+		Type: "Microservice",
+		Data: CronJobData,
+	}, nil
+}
+
+//create a data object that contains name, labels, CronJobime, namespace, version and image of the CronJob and returns as KubernetesObject
+func CronJobDataMapping(CronJob batchv1.CronJob) (*mapper.KubernetesObject, error) {
+	var CronJobData map[string]interface{}
+	CronJobData = make(map[string]interface{})
+	var version string
+	var CronJobVersion string
+	var CronJobVersionShort string
+	if _, ok := CronJob.ObjectMeta.Labels["app.kubernetes.io/version"]; ok {
+		CronJobVersion = CronJob.ObjectMeta.Labels["app.kubernetes.io/version"]
+		CronJobVersionShort = CronJob.ObjectMeta.Labels["app.kubernetes.io/version"]
+	} else {
+		version = ""
+		CronJobVersion = "unknown version"
+		CronJobVersionShort = "unknown"
+	}
+	CronJobImage := strings.Split(CronJob.Spec.JobTemplate.Spec.Template.Spec.Containers[0].Image, ":")[0]
```

Comment:
Can we rely on the Containers slice always having elements?

Reply:
Let me verify that

Reply:
Yeah containers are always required otherwise you will get an error creating the resource so it should be fine from my view.